src/live/embed.rs

```diff
@@ -477,8 +498,14 @@ impl KargEmbedAreas {
                 self.length
             );
         }
-        let mut contents = vec![b'#'; self.length];
-        contents[..formatted.len()].copy_from_slice(formatted.as_bytes());
+
+        let filler = self.filler.unwrap_or('#');
+        let size = match self.arch.as_ref() {
+            Some(arch) if arch == "s390x" => formatted.len() - 1, // s390x ISO: kargs line stored within cdboot.img file ends with zeros and doesn't contain '\n'
```

Comment:
hmm. for some reason I thought the reason we encoded the `filler` was so we could drop encoding the architecture: https://github.com/coreos/coreos-assembler/pull/3527#discussion_r1263118197

but we still need to?

Reply:
as i got from @bgilbert comments it'd be good to have the `filler`:
> non-s390x always wants # and s390x always wants \0, which isn't guaranteed to always be true

but we still need to skip `\n` for s390x, so `arch` field is also required

Reply:
could we just encode this as `endbyte` or something? it feels like basically the same thing as the argument for why to encode `filler` versus relying on the architecture like you were doing before.

Reply:
I agree with Dusty that we should avoid having an `arch` field.  See https://github.com/coreos/coreos-assembler/pull/3527#discussion_r1267632070.